daemons/fenced/fenced_commands.c

```diff
@@ -390,15 +390,15 @@ stonith_device_execute(stonith_device_t * device)
         return TRUE;
     }
 
-    if(pcmk__str_eq(device->agent, STONITH_WATCHDOG_AGENT, pcmk__str_casei)) {
-        if(pcmk__str_eq(cmd->action, "reboot", pcmk__str_casei)) {
-            pcmk__panic(__func__);
-            goto done;
-
-        } else if(pcmk__str_eq(cmd->action, "off", pcmk__str_casei)) {
-            pcmk__panic(__func__);
-            goto done;
-
+    if (pcmk__str_any_of(device->agent, STONITH_WATCHDOG_AGENT,
+                         STONITH_WATCHDOG_AGENT_INTERNAL, NULL)) {
```

Comment:
This will allow a newer node to execute a request initiated on a different node regardless of whether it is older or newer, but older nodes will only look for AGENT_INTERNAL, right? Do we need to make sure we only send remote requests using AGENT_INTERNAL?

Reply:
Only case where we support nodes to stay on older version permanently is in case of remote-nodes - right?
Remote-nodes are not touched here as they will still send their query as before and the difference in handling is done on the cluster-nodes.
As long as you don't add explicit configuration for the watchdog-fencing-resource we will just send AGENT_INTERNAL stuff including the corresponding class. As on other rolling upgrades the incompatible config-change is just done after all nodes are on the new version. Thus I don't think we need extra pre-cautions here.

Reply:
Bump CRM_FEATURE_SET to get that behavior, otherwise cluster nodes can stay on the old version as well.

With most config changes, we rely on the schema to prevent users from using the new feature before all nodes are upgraded, since the DC will be the oldest node and won't have the new schema to upgrade to. That doesn't work for changes in attribute values though, we just have to rely on users being careful. As long as we bump the feature set, that should be acceptable.